File: libs/contracts/Dto/DoctorDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class DoctorDto
    {
        public int Id { get; set; }
        public int BranchId { get; set; }
        public string? BranchName { get; set; }
        public int UserId { get; set; }
        public string? DoctorName { get; set; }
        public int? DepartmentId { get; set; }
        public string? DepartmentName { get; set; }
        public string? Specialization { get; set; }
        public string? LicenseNumber { get; set; }
        public decimal? ConsultationFee { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset? CreatedAt { get; set; }
        public int? CreatedBy { get; set; }

        public string? createdUser { get; set; }
        public DateTimeOffset? UpdatedAt { get; set; }
        public int? UpdatedBy { get; set; }

        public string? updateUser { get; set; }
    }
}

File: apps/HSMS-fe/src/app/models/Doctor.dto.ts
export interface CreateDoctorDto {
  branchId: number;
  userId: number;
  departmentId?: number;
  specialization?: string;
  licenseNumber?: string;
  consultationFee?: number;
  isActive: boolean;
  createBy?: number;
}

export interface UpdateDoctorDto {
  id: number;
  branchId: number;
  departmentId?: number;
  specialization?: string;
  licenseNumber?: string;
  consultationFee?: number;
  isActive: boolean;
  updateBy?: number;
}


export interface DoctorDto {
  id: number;
  branchId: number;
  branchName?: string;
  userId: number;
  doctorName?: string;
  departmentId?: number;
  departmentName?: string;
  specialization?: string;
  licenseNumber?: string;
  consultationFee?: number;
  isActive: boolean;
  createdAt?: string;
  createdBy?: number;
  createdUser?: string;
  updatedAt?: string;
  updatedBy?: number;
  updateUser?: string;
}